ekncontent/ekncontent/eknc-query-object.c

```diff
@@ -510,24 +507,40 @@ get_terms (const gchar *query)
   return terms;
 }
 
-static gchar *
-get_exact_title_clause (EkncQueryObject *self, gchar **terms)
+static XapianQuery *
+get_exact_title_clause (EkncQueryObject *self,
+                        XapianQueryParser *qp,
+                        char **terms,
+                        GError **error_out)
 {
+  g_autoptr(GError) error = NULL;
   g_autofree gchar *joined = g_strjoinv ("_", terms);
-  g_autofree gchar *prefixed = g_strconcat (XAPIAN_PREFIX_EXACT_TITLE, joined, NULL);
-
-  if (self->mode == EKNC_QUERY_OBJECT_MODE_DELIMITED)
-    return g_steal_pointer (&prefixed);
+  g_autofree gchar *prefixed = g_strconcat (QUERY_PARSER_PREFIX_EXACT_TITLE, joined, NULL);
 
   /* Combine the term with a wild-carded version if search mode is incremental */
-  return g_strdup_printf ("(%s%s%s*)", prefixed, XAPIAN_OP_OR, prefixed);
+  XapianQueryParserFeature flags = XAPIAN_QUERY_PARSER_FEATURE_DEFAULT;
+  if (self->mode == EKNC_QUERY_OBJECT_MODE_INCREMENTAL)
+    flags |= XAPIAN_QUERY_PARSER_FEATURE_PARTIAL;
+
+  g_autoptr(XapianQuery) retval =
+    xapian_query_parser_parse_query_full (qp, prefixed, flags, "", &error);
```

Comment:
If I follow the code, you could avoid having to create `prefixed` if you specify a default prefix like this:

```
  xapian_query_parser_parse_query_full (qp, joined, flags, XAPIAN_PREFIX_EXACT_TITLE, &error);
```

Note this would need `XAPIAN_PREFIX_EXACT_TITLE` not `QUERY_PARSER_PREFIX_EXACT_TITLE`.